crates/bevy_ui/src/lib.rs

```diff
@@ -75,6 +78,97 @@ impl Default for UiScale {
     }
 }
 
+/// General representation of progress between two values.
+#[derive(Debug)]
+pub struct Progress<T>
+where
+    T: Send + Sync + Copy,
+{
+    /// The minimum value that the progress can have, inclusive.
+    min: T,
+    /// The maximum value that the progress can have, inlucsive.
+    max: T,
+    /// The current value of progress.
+    value: T,
+}
+
+impl<T: Send + Sync + Copy> Progress<T>
+where
+    T: PartialOrd<T>,
+{
+    /// Creates a new progress using a value, min and max.
+    ///
+    /// The value must be within min and max.
+    /// Will panic if value is out-of-bounds.
+    pub fn new(value: T, min: T, max: T) -> Self {
+        Self::from_range(value, min..=max)
+    }
+
+    /// Creates a new progress using a value and a range.
+    ///
+    /// The value must be within the bounds of the range.
+    /// Will panic if value is out-of-bounds.
+    pub fn from_range(value: T, range: RangeInclusive<T>) -> Self {
+        if range.contains(&value) {
+            Self {
+                value,
+                min: *range.start(),
+                max: *range.end(),
+            }
+        } else {
+            panic!("Tried creating progress with value out of bounds.");
+        }
+    }
+
+    /// Gets the min bound of the progress.
+    pub fn get_min(&self) -> T {
```

Comment:
Bevy uses `get_` for fallible getters. For getters that can't fail no prefix is used.